common/src/main/scala/org/apache/celeborn/common/meta/WorkerInfo.scala

```diff
@@ -34,12 +34,19 @@ class WorkerInfo(
     val pushPort: Int,
     val fetchPort: Int,
     val replicatePort: Int,
-    val diskInfos: util.Map[String, DiskInfo],
-    val userResourceConsumption: util.Map[UserIdentifier, ResourceConsumption],
-    var endpoint: RpcEndpointRef) extends Serializable with Logging {
+    _diskInfos: util.Map[String, DiskInfo],
+    _userResourceConsumption: util.Map[UserIdentifier, ResourceConsumption]) extends Serializable
+  with Logging {
   var unknownDiskSlots = new java.util.HashMap[String, Integer]()
   var networkLocation = "/default-rack"
   var lastHeartbeat: Long = 0
+  val diskInfos =
+    if (_diskInfos != null) JavaUtils.newConcurrentHashMap[String, DiskInfo](_diskInfos) else null
```

Comment:
how about assigning an empty collection instead of `null`?

Reply:
No strong objections, but I think NPE is evil, would try my best to avoid null propagation.

Reply:
Got your point. I just feel that it's hard to totally avoid null, especially for non-Collection members without or with a heavy default constructor.

Reply:
Most of the cases are performance insensitive. For non-collection cases, we can use `Option[T]` with the default value `None`.